Makefile

```diff
@@ -32,11 +32,21 @@ dev: env
 
 
 ## Runs unit tests.
-test:
+test: lint
 	@echo "Running unit tests."
 	$(PIPENV_CMD) run tox
 
 
+## Lint source files.
+lint: black flake8 mypy
+
+
+## Reformat source files with black.
+format:
+	@echo "Running black to format source files."
+	$(PIPENV_CMD) run black src test
```

Comment:
I don't have an strong opinion here. I would like to know what other think about this.

Would it make it clearer to the reader if we include the config file as a command parameter here? Like:
`$(PIPENV_CMD) run black --config pyproject.toml src test`

From [black docs](https://black.readthedocs.io/en/stable/pyproject_toml.html#pyproject-toml) it is clear that when the command is run, it will look for a `pyproject-toml` file on common base directory but not sure it this is something that everybody will think about by just looking at the command.

Reply:
That's a great idea to explicitly call out the config file. I think the flake8 config can be done in there as well, I'll check.

Reply:
OK, done in the latest commit - explicit config file for black. For flake8, it *can* use pyproject.conf, however the flake8 config is not compatible with black so adding flake8 config causes black to fail reading the config! So, instead I added the flake8 configuration to tox.ini - which seems a little strange, but is apparently OK (documented at flake8).

Alternatively, we can move flake8 config out to it's own file, but it being in tox.ini didn't seem too bad. Thoughts?

Reply:
I've seen the flake8 config in the tox.ini file in different projects. I prefer that instead of creating a new config file.

Reply:
OK, great, thanks!